README.md
# SEO robot pro průběžné každodenní testování webů a monitoring

Denně se na webech, kde pracujete na SEO může měnit fůra věcí, protože obvykle nemáte pod kontrolou všechny úpravy na webu - běžné releasy, hot fixy, atp.
SEO robot umí otestovat vámi definovaná pravidla pro typové stránky webu jako:
* Vrací URL správný HTTP kód?
* Je na webu správně nastaven titulek, popisek, meta robots, OG tagy, atd. ?
* Je soubor robots.txt dostupný, je tam, kde má být a nezměnil se?
* Je v HTML takový obsah, jaký tam má být?

SEO robot projde web a otestuje vše, co mu zadáte a když narazí na chybu, tak vám pošle email:

![Náhled emailu](https://www.hlavinka.cz/seoskoleni/nahled-emailu.png "Title")

Tak se můžete spolehnout na to, že vám pod rukama nikdo nemění technické SEO a obsah stránek.


## Jak správně nastavit proces testování SEO?
### Testy v testovacím prostředí
Už teď pravděpodobně testujete web předtím, než nasadíte novou verzi do provozu. Testujete, jestli fungují všechny stránky, stavy aplikace, formuláře, emaily, jestli web vypadá dobře i v IE, atd. Na SEO se většinou nemyslí a netestuje se, jestli se při úpravách neudělala chyba, která rozbije už hotové technické nastavení SEO.

**Naučte vaše programátory testovat na testu i SEO** (klidně i jinak, než SEO robotem).

SEO robot může běžet na testovacím prostředí a před nasazením jen ručně, nebo automaticky spusťte předem definované testy, které otestují SEO na TESTovací verzi webu. Pokud se stane nějaká chyba - pak pošle email SEO konzultantovi a rovnou i odpovědné osobě za web - produktový manažer, webmaster, kdokoliv, kdo může odhalenou chybu začít opravovat (případně rozhodnout, že je to maličkost, která nemá vliv na nasazení.

Odhalení chyby na testu je výrazně lepší, než chybu najít až po nasazení v provozu.


### Testy v produkčním prodstředí
Druhý typ testů v produkci je průběžný a spouští se automaticky Cronem každý den (v Seznamu to mám nastaveno každý den co 20 minut). Tím  se nejpozději do 20 minut po změně na webu dozvím, že se změnilo něco, co může ovlivnit SEO. To je relativně hodně rychlé a snižuje to případné problémy.


# Instalace
## Požadavky na prostředí:
* Webhosting, nebo
* Linux server s PHP a Cron (třeba VPS)


## Jednoduchá instalace pro začátečníky:
* Stáhněte .ZIP - vpravo nahoře
* Rozbalte SEO robota do adresáře, kde máte webové aplikace
* Nastavte SEO robota tak, aby byl dostupný na URL dostupné z Cron - může být i veřejně dostupný, ale pak je lepší nastavit přístup jen z Cron a z vašich IP, aby se tam nedostal někdo, kdo tam nemá co dělat :)
* Nastavte práva k souborům (chmod)


## Instalace pro profíky
* programátoři, webaři a jiní profíci, vy ode mne návod k instalaci nepotřebujete ;-)


# Nastavení
## 1. Nastavení emailů a SMTP
Nastavení najdete v **/settings/settings.php v poli $generalSettings**.
Při chybě posílá SEO robot email - nejprve nastavíme SMTP, aby se emaily mohly posílat. Pro posílání emailů používá SEO robot knihovnu PHPmailer - tzn. nastavení SEO robota je rochu ořezané nastavení PHPmailera.
Příklad nastavení pro Gmail (to je takové nejvíce specifické):
```
'smtpDebugLevel' => 4, // hodnota 4 je nejvíce ukecaná a řekne vám toho hodně. Po otestování nastavte zpět na 0.
'emailHost' => 'smtp.gmail.com',
'emailPort' => 587,
'emailSMTPSecure' => 'tls',
'emailSMTPAuth' => true,
'emailUsername' => 'nove-vytvoreny-email-pro-posilani-notifikaci@gmail.com',
'emailPassword' => 'heslodoemailu', // do budoucna bude xoauth2, tzn. heslo do emailu nebude nikde vidět
```
! Pro odesílání emailů je nejlepší vytvořit si úplně nový email na Seznamu, nebo Gmailu.


## 2. Nastavení SEO testů


### 2.1 Založení nového projektu
Nový projekt vytvoříte v **/settings/settings.php v poli $testsSettings**.
Stačí přidat nové pole s názvem služby do pole $testsSettings. Název služby je důležitý - podle něj budete pak test spouštět.

```
'sbazar.cz' => // 1. ID projektu
    array( 	
        'testRules' => './settings/tests/sbazar.txt', // 2. cesta k souboru se SEO testy
        'robotsTxtURL' => 'https://www.sbazar.cz/robots.txt', // 3. cesta k robots.txt na serveru
        'robotsTxtFile' => './settings/robots.txt/sbazar.txt', // 4. cesta k robots.txt uloženému lokálně
        'curl_useragent' => 'SEO test', // 5. user-agent
        'email' => 'jaroslav.hlavinka@firma.seznam.cz', // 6. adresát
    ),
```
Detailně vysvětleno:
1. **ID projektu** - musí být unikátní pro každý projekt. Tímto ID se spouští projekty např. /index.php?id=sbazar.php
2. **cesta k souboru se SEO testy** - cesta, kde má SEO robot hledat soubor s předdefinovanými SEO testy
3. **cesta k robots.txt na serveru**
4. **cesta k robots.txt uloženému lokálně** - pro ověření, jestli se robots.txt na serveru nezměnil. Při tvorbě testu stáhněte soubor robots.txt z webu a uložte ho SEO robotovi do adresáře /settings/robots.txt/
5. **user-agent** - toho můžete měnit a monitorovat případné specifické chování pro jiné useragenty -Facebot, GoogleBot mobile, SeznamBot, atd. Pro test s jiným user-agentem si založte nový projekt
6. **adresát** - komu se pošle email v případě chyby v testech


### 2.2 Založení SEO testů
SEO testy se zapisují jednoduše do textového souboru a SEO robot je čte řádek po řádku.

**Prázdné řádky a řádky začínající znakem #** SEO robot přeskakuje. # můžete používat pro vaše interní poznámky k testům.

Jak vypadá zápis:
1. **URL (povinný parametr)** - tuto URL SEO robot stáhne a její HTTP kód a HTML bude dále testovat
2. **číslo HTTP statu kódu (povinný parametr)** - pokud je číslo HTTP kódu, který výše uvedená URL vrátí, stejné, tak URL testem HTTP kódu prošla. Sem dávejte pouze číslo. Dobře je "301", špatně je "301 Moved Permanently".
3. **definice SEO testů (nepovinný parametr)** - samotná pravidla testů. Viz další kapitola

Příklad: stáhnou se 2 URL a provedou se 4 testy - 2 testy HTTP kódu a 2 testy obsahu homepage:
```
# testy na homepage
https://www.zbozi.cz/?_escaped_fragment_=
200
plaintext;;//*[@id="homepage-discounts-title"];;0;;Zboží ve slevě
plaintext;;//title;;0;;Zboží.cz • Tisíce obchodů na jednom místě s možností srovnání cen

# naprosto nedůležitý komentář

# testy HTTP kódu
https://www.zbozi.cz/iphone
200

```


#### 2.2.1 definice SEO testů
U každé URL může být 0 až X testů. Každý na vlastním řádku. Můžete tak pro jednu URL otestovat zároveň titulek, meta description a obsah stránky. Ale také nemusíte uvést žádný SEO test - například, když chcete změřit přesměrování a HTTP chyby 4XX a 5XX.

Definice SEO testů se oddělují dvěma středníky ;; a mají přesný zápis, který je potřeba dodržet:
1. **první parametr** říká, co se má vytáhnout z následujícího pravidla. Tady jsou pevně dané možnosti:
    * **plaintext** - vybere jen text elementů bez HTML tagů
    * **content** - vybere obsah meta tagů
    * **href** - vybere obsah atributu href například v odkazech, nebo rel canonical
    * **plaintextContains** - zjistí, jestli plaintext HTML tagu zadaného pomocí Xpath obsahuje zadaný text
    * **contentContains** - zjistí, jestli content HTML tagu zadaného pomocí Xpath obsahuje zadaný text
    * **hrefContains** - zjistí, jestli atribut href HTML tagu zadaného pomocí Xpath obsahuje zadaný text
2. **druhý parametr** je XPath cesta k elementu, který se má otestovat
3. **pořadí elementu vybraného v XPath** - vámi vybraný element bude asi nejčastěji 0 (první v pořadí)
4. **hodnota**, kterou očekáváte a má se otestovat

Příklady:
1. v TITLE je přesně toto: "Renault Trafic. - Sbazar.cz":
    * ```plaintext;;//title;;0;;Renault Trafic. - Sbazar.cz```
2. v META description je "Inzerát Renault Trafic. v okrese Praha-východ, cena 0 Kč, od ciciolina.sicher na Sbazar.cz":
    * ```content;;meta[name="description"];;0;;Inzerát Renault Trafic. v okrese Praha-východ, cena 0 Kč, od ciciolina.sicher na Sbazar.cz```
3. META description obsahuje "Inzerát Renault Trafic. v okrese Praha-východ":
    * ```contentContains;;meta[name="description"];;0;;Inzerát Renault Trafic. v okrese Praha-východ```
4. v META robots je "noindex,nofollow"
    * ```content;;meta[name="robots"];;0;;noindex,nofollow```
5. META robots je prázdný, nebo na stránce vůbec není
    * ```content;;meta[name="robots"];;0;;```
6. REL canonical je přesně: "https://www.sbazar.cz"
    * ```href;;link[rel="canonical"];;0;;https://www.sbazar.cz```
7. REL next je: "https://www.sbazar.cz"
    * ```href;;link[rel="next"];;0;;https://www.sbazar.cz```
8. REL prev obsahuje: "/3"
    * ```hrefContains;;link[rel="prev"];;0;;/3```
9. H1 je "Renault Trafic"
    * ```plaintext;;h1;;0;;Renault Trafic.```
10. V 3. tagu LI v XPath "#box8 > div > div > ul > li" je "něco"    
    * ```plaintext;;#box8 > div > div > ul > li;;2;;něco```
11. V tagu s ID "localityNameDesktop" je "Praha"
    * ```plaintextContains;;#localityNameDesktop;;0;;Praha```



**Tip:** pro první spuštění si udělejte v testech záměrnou chybu, aby se email poslal a vy ho viděli


### 2.3 Spouštění testů
Každá ze služeb se spouští zvláště. Nejsou nijak zřetězeny.

Já doporučuji testy spouštět takto často:
* velký web, kde se pořád něco děje (eshop, atp.) - co 20 minut
* malý web, kde se nic neděje - stačí jednou denně (pokud to vůbec má smysl monitorovat SEO robotem)
* střední web - někde mezi 20 minutami a jedním dnem

Spuštění uděláte nejlépe pomocí Cron.
* Pro hostované weby: V nastavení hostingu určitě máte naklikávátko Cron - naklikejte každou
* Pro Linux:
```
0,20,40 * * * * root curl https://seo.dev.dszn.cz/seorobot/index.php?id=sbazar.cz >> /dev/null 2>&1
2,22,42 * * * * root curl https://seo.dev.dszn.cz/seorobot/index.php?id=zbozi.cz >> /dev/null 2>&1
4,24,44 * * * * root curl https://seo.dev.dszn.cz/seorobot/index.php?id=firmy.cz >> /dev/null 2>&1
6,26,46 * * * * root curl https://seo.dev.dszn.cz/seorobot/index.php?id=firmy-mobilni.cz >> /dev/null 2>&1


```

**Tip:** Jednou za nějaký čas si spusťte testovací skript napříč všemi vašimi weby, který projde z každého webu dvě URL - jednu bez chyby, druhou se záměrnou chybou. Takový monitoring funkčnosti SEO robota - abyste věděli, že stále funguje a nežili v milé nevědomosti, že když nechodí emaily, tak je to dobře. Obvykle není :-)


CHANGELOG.md
# Changelog
Co se na SEO robotovi změnilo?

## [1.1.3] - 2018-06-22
### Novinky
- Lze nastavit více emailových adres pro odeslání reportu. Oddělovač je středník ";"


## [1.1.2] - 2018-03-16
### Novinky
- Nyní umí rovnou rozbalit gzipped stránky v CURLu.


## [1.1.1] - 2018-03-15
### Novinky
- Nová pravidla pro porovnávání: plaintextContains, hrefContains, contentContains - test projde, když Xpath obsahuje zadaný text (to je vhodné pro automaticky generované popisky, titulky, odkazy, atd.)


app/app.php
<?php

require './vendor/autoload.php';
use Sunra\PhpSimple\HtmlDomParser;
use PHPMailer\PHPMailer\PHPMailer;
use PHPMailer\PHPMailer\Exception;
$log .= '';

// test if the testid is defined in settings file - settings.php
if ( !array_key_exists( $testid, $testsSettings ) ) {
	die('V tabulce $testsSettings v souboru settings.php není definován test pro toto testid: "'. $testid .'".');
} else {
	$settings = $testsSettings[ $testid ];
}

// set user agent
if ( array_key_exists( 'curl_useragent', $settings ) ) {
	define('DEFAULT_USER_AGENT', $settings['curl_useragent']);
} else if ( array_key_exists( 'curl_useragent', $generalSettings ) ) {
	define('DEFAULT_USER_AGENT', $generalSettings['curl_useragent']);
} else {
	define('DEFAULT_USER_AGENT', '');
}

$log .= logger('Začátek testu: ' . date("H:i:s Y-m-d"), 'info' );
$log .= logger('Služba: ' . $testid, 'info' );

// start robots.txt test
if ( array_key_exists( 'robotsTxtURL', $settings ) && array_key_exists( 'robotsTxtFile', $settings ) &&
	$settings['robotsTxtURL'] != '' && $settings['robotsTxtFile'] != '' ) {
		$log .= logger('Testy robots.txt', 'boldInfo');
		$response = downloadURL( $settings['robotsTxtURL'] );
		if ( $response['statusCode'] != '200' ) {
				$log .= logger('Robots.txt na adrese ' .$settings['robotsTxtURL'].
									 ' vrátil chybu ' . $response['status'], 'error' );
		} else {
				$log .= logger('Robots.txt je dostupný.');
		}

		// test if robots.txt is the same as in the file
			$robotsFileString = file_get_contents( $settings['robotsTxtFile'] );
			if ( $robotsFileString === FALSE ) {
					$log .= logger('Chyba při čtení souboru "' .$settings['robotsTxtFile'].
					'". Například protože neexistuje, nebo nemáte práva jej číst.', 'error');
			} else {
        if ( $response['html'] === $robotsFileString) {
            $log .= logger('Robots.txt je stejný jako testovací.');
        } else {
            $log .= logger('Robots.txt na '. $settings['robotsTxtURL']
            .' je jiný, než testovací.', "error");
        }
      }

} else {
    $log .= logger('Testy robots.txt nejsou definovány, takže neproběhly.', "notice");
}

// start SEOtests
if ( array_key_exists( 'testRules', $settings ) && $settings['testRules'] != '' ) {
		$log .= logger('Testy typových stránek', 'boldInfo');

    $testsFileArray = file( $settings['testRules'], FILE_SKIP_EMPTY_LINES);
		if ( $testsFileArray === FALSE ) {
				$log .= logger('Chyba při čtení souboru "' .$settings['testRules'].
				'". Například protože neexistuje, nebo nemáte práva jej číst.', 'error');
		} else {
        $current = array('url' => '', 'statusCode' => '', 'html' => '');


        foreach ($testsFileArray as $test) {
          $test = trim($test);
            if( $test == '' || substr( $test, 0, 1 ) === "#" ) { // skip commented and empty lines
                continue;

            } else if( substr( $test, 0, 4 ) == "http" ) { // is URL
                $response = downloadURL( $test );
                $current['url'] = $test;
                $current['statusCode'] = $response['statusCode'];
                $current['html'] = $response['html'];
                $log .= logger('Testy pro URL '.$test. ':', "boldInfo");

            } else if ( preg_match("/^[0-9]+$/", $test ) ) { // is status Code
                if ( $current['statusCode'] == $test ) {
                    $log .= logger('Správný HTTP kód '.$test);
                } else {
                    $log .= logger('Špatný HTTP kód. Má být HTTP '.
                            $test .', ale vrátil se HTTP '. $current['statusCode'], 'error');
                }
            } else if ( preg_match("/^(href|hrefContains|plaintext|plaintextContains|content|contentContains).*/", $test )) { // tests
                $a = explode(';;', $test);
                $html = HtmlDomParser::str_get_html( $current['html'] );
                $haystack = '';
                switch ($a[0]) {
                  case 'href':
	                    $haystack = trim ($html->find($a[1], $a[2])->href);
											$needle = $a[3];
	                    if( $haystack == $needle) {
	                        $log .= logger('v "'. $a[1] . ' [' . $a[2] .']" je správně "'. $needle .'".' );
	                    } else {
	                        $log .= logger('v "'. $a[1] . '[' . $a[2] .']" má být "'. $needle .'", ale na stránce je  "'. $haystack . '"', 'error');
	                    }
	                    break;


									case 'hrefContains':
											$haystack = trim ($html->find($a[1], $a[2])->href);
											$needle = $a[3];
											if( strpos($haystack, $needle) !== FALSE ) {
													$log .= logger('v "'. $a[1] . ' [' . $a[2] .']" správně obsahuje "'. $needle .'".' );
											} else {
													$log .= logger('v "'. $a[1] . '[' . $a[2] .']" má obsahovat "'. $needle .'", ale na stránce je  "'. $haystack . '"', 'error');
											}
											break;


                  case 'plaintext':
	                    $haystack = trim ($html->find($a[1], $a[2])->plaintext);
											$needle = $a[3];
	                    if( $haystack == $needle) {
	                        $log .= logger('v "'. $a[1] . '[' . $a[2] .']" je správně "'. $needle .'".' );
	                    } else {
	                        $log .= logger('v "'. $a[1] . '[' . $a[2] .']" má být "'. $needle .'", ale na stránce je  "'. $haystack . '"', 'error');
	                    }
	                    break;


									case 'plaintextContains':
	                    $haystack = trim ($html->find($a[1], $a[2])->plaintext);
											$needle = $a[3];
	                    if( strpos($haystack, $needle) !== FALSE ) {
	                        $log .= logger('v "'. $a[1] . '[' . $a[2] .']" správně obsahuje "'. $needle .'".' );
	                    } else {
	                        $log .= logger('v "'. $a[1] . '[' . $a[2] .']" má obsahovat "'. $needle .'", ale na stránce je "'. $haystack . '"', 'error');
	                    }
	                    break;


                  case 'content':
	                    $haystack = trim ($html->find($a[1], $a[2])->content);
											$needle = $a[3];
	                    if( $haystack == $needle) {
	                        $log .= logger('v "'. $a[1] . ' [' . $a[2] .']" je správně "'. $needle .'".' );
	                    } else {
	                        $log .= logger('v "'. $a[1] . '[' . $a[2] .']" má být "'. $needle .'", ale na stránce je "'. $haystack . '"', 'error');
	                    }
	                    break;


									case 'contentContains':
	                    $haystack = trim ($html->find($a[1], $a[2])->content);
											$needle = $a[3];
	                    if( strpos($haystack, $needle) !== FALSE ) {
	                        $log .= logger('v "'. $a[1] . ' [' . $a[2] .']" správně obsahuje "'. $needle .'".' );
	                    } else {
	                        $log .= logger('v "'. $a[1] . '[' . $a[2] .']" má obsahovat "'. $needle .'", ale na stránce je "'. $haystack . '"', 'error');
	                    }
	                    break;

                  default:
                    break;
                }
            }
        }
    }

} else {
     $log .= logger('Testy nejsou definovány, takže neproběhly.', "notice");
}

$log .= logger('<br><br>Konec testu: ' . date("H:i:s Y-m-d"), 'info' );

// if there is even one error, we send email
if($hasError === TRUE) {
  if ( array_key_exists( 'emailUsername', $generalSettings ) && array_key_exists( 'emailPassword', $generalSettings)
  && array_key_exists( 'email', $settings)) {

			$mail = new PHPMailer;
			$mail->CharSet = "UTF-8";
			$mail->isSMTP();
			$mail->SMTPDebug = $generalSettings['smtpDebugLevel']; // 0
			$mail->Host = $generalSettings['emailHost'];
			$mail->Port = $generalSettings['emailPort'];
			$mail->SMTPSecure = $generalSettings['emailSMTPSecure'];
			$mail->SMTPAuth = $generalSettings['emailSMTPAuth'];
			$mail->Username = $generalSettings['emailUsername'];
			$mail->Password = $generalSettings['emailPassword'];
			$mail->setFrom($generalSettings['emailFrom']);
			$mail->addReplyTo($generalSettings['emailFrom']);

			// když je více emailů
			if ( strpos( $settings['email'], ';') !== false ) {
					$emails = explode(';', $settings['email'] );
					foreach ($emails as $email) {
							$mail->addAddress($email);
					}
			}	else {
					$mail->addAddress($settings['email']);
			}
			$mail->Subject = 'SEO test se nezdařil - '. $testid;
			$mail->msgHTML($log);
			$mail->AltBody = strip_tags($log);
			$mail->smtpConnect([
					'ssl' => [
							'verify_peer' => false,
							'verify_peer_name' => false,
							'allow_self_signed' => true
					]
			]);
      //send the message, check for errors
      if (!$mail->send()) {
          echo "Chyba - email nebyl poslán: " . $mail->ErrorInfo;
          if (!$mail->send()) {
              echo "Chyba - email nebyl poslán: " . $mail->ErrorInfo;
          } else {
              echo "Email odeslán!";
          }
      } else {
          echo "Email odeslán!";
      }
  }
}


app/functions.php
<?php

/**
 * Downloads HTML response and status code of URL.
 *
 * @param string $url URL to download
 * @param string $useragent Optional. Useragent of the request. Default can be changed in settings file
 * @return array htmlContent and statusCode.
 */
function downloadURL($url, $useragent = DEFAULT_USER_AGENT){
	$ch = curl_init($url);

	curl_setopt($ch, CURLOPT_USERAGENT, $useragent);
	curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
	curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);
	curl_setopt($ch, CURLOPT_HEADER, false);
	curl_setopt($ch, CURLOPT_FOLLOWLOCATION, false);
	curl_setopt($ch, CURLOPT_ENCODING , "");

	$response = curl_exec($ch);
	$httpCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);

	curl_close($ch);

	return array('statusCode' => $httpCode, 'html' => $response );
}


/**
 * logging errors
 *
 * @param string $text Description of an error
 * @param string $type
 * @return html string
 */
function logger($text, $type = "ok") {
		$htmlLog = '';
		if ( $type == "ok" ) {
				$htmlLog = "<span style='color:lime;font-weight:bold'>OK: </span><span>".$text."</span><br>";
		} else if ( $type == "error") {
				$htmlLog = "<span style='color:red;font-weight:bold'>CHYBA: </span><span>".$text."</span><br>";
				$GLOBALS['hasError'] = TRUE;
		} else if ( $type == "notice") {
				$htmlLog = "<span style='color:orange;font-weight:bold'>POZOR: </span><span>".$text."</span><br>";
		} else if ( $type == "boldInfo") {
				$htmlLog = "<br><b>".$text."</b><br>";
		} else if ( $type == "info") {
				$htmlLog = "".$text."<br>";
		}
		echo $htmlLog;
		return $htmlLog;
}



function getRowsFromFile($filename){
  $data = file("./urls/".$filename, FILE_SKIP_EMPTY_LINES);
  $unused = array_splice($data,LIMITPERDAY-1);
  // rewrite file
  $string_data = implode($unused);
  file_put_contents("./urls/".$filename, $string_data);

  return array_splice($data,0,LIMITPERDAY);
}


settings/settings.php
<?php

$testsSettings = array(

	'sbazar.cz' =>
		array( 	'robotsTxtFile' => './settings/robots.txt/sbazar.txt',
			'robotsTxtURL' => 'https://www.sbazar.cz/robots.txt',
			'testRules' => './settings/tests/sbazar.txt',
			'curl_useragent' => 'SEO test',
			'email' => 'email-kam-poslat-report@seznam.cz', // středník odděluje více emailů
		 ),

	// same test as previous, but with different useragent for testing "hacks for googlebot mobile"
	'zbozi.cz-googlebot-mobile' =>
		array( 	'robotsTxtFile' => '/testfiles/robots.txt/zbozi.txt',
				'curl_useragent' => 'Mozilla/5.0 (Linux; Android 6.0.1; Nexus 5X Build/MMB29P) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/41.0.2272.96 Mobile Safari/537.36 ( ompatible; Googlebot/2.1; +http://www.google.com/bot.html)',
		 ),

	// sample configuration for "default.cz" test
	'default.cz' =>
		array( 	'robotsTxtFile' => '/testfiles/robots.txt/default.txt',
				'testRules' => '/testfiles/tests/default.txt',
				'curl_useragent' => 'SEO test',
		 ),

 );



/**
*	default settings
*
*/

$generalSettings = array(
				'curl_useragent' => 'SEO testing robot - made with ♥ by Jaroslav Hlavinka jaroslav@hlavinka.cz',
				'debug' => TRUE,
				'emailHost' => 'smtp.gmail.com',
				'emailPort' => 587,
				'emailSMTPSecure' => 'tls',
				'emailSMTPAuth' => true,
				'emailUsername' => 'email@gmail.com',
				'emailPassword' => 'heslodoemailu',
				'emailFrom' => 'vrazda-smrt-zabiti@seorobot.dev',
				'smtpDebugLevel' => 0, // 2 or 4 when debugging
			);
$log = '';
// error_reporting(E_ERROR | E_WARNING | E_PARSE | E_NOTICE);
//error_reporting(E_ALL);
error_reporting(E_ERROR);
